src/components/forms/controls/Checkbox/CheckboxTri.stories.tsx

```diff
@@ -0,0 +1,94 @@
+/* Copyright (c) Fortanix, Inc.
+|* This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. If a copy of
+|* the MPL was not distributed with this file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import type { Meta, StoryObj } from '@storybook/react';
+
+import * as React from 'react';
+
+import { CheckboxTri } from './CheckboxTri.tsx';
+
+
+type CheckboxTriArgs = React.ComponentProps<typeof CheckboxTri>;
+type Story = StoryObj<CheckboxTriArgs>;
+
+export default {
+  component: CheckboxTri,
+  parameters: {
+    layout: 'centered',
+  },
+  tags: ['autodocs'],
+  argTypes: {
+  },
+  args: {
+    'aria-label': 'Test checkbox',
+  },
+  decorators: [
+    Story => <form onSubmit={event => { event.preventDefault(); }}><Story/></form>,
+  ],
+  render: (args) => <CheckboxTri {...args}/>,
+} satisfies Meta<CheckboxTriArgs>;
+
+
+export const CheckboxTriStandard: Story = {
+  name: 'CheckboxTri',
+  args: {
+    defaultChecked: 'indeterminate',
+  },
+};
+
+export const CheckboxTriDisabled: Story = {
+  name: 'CheckboxTri [disabled]',
+  args: {
+    disabled: true,
+    defaultChecked: 'indeterminate',
+  },
+};
+
+export const CheckboxTriFocusedIndeterminate: Story = {
+  name: 'CheckboxTri [focused]',
+  args: {
+    className: 'pseudo-focus-visible',
+    defaultChecked: 'indeterminate',
+  },
+};
+
+export const CheckboxTriFocusedDisabled: Story = {
+  name: 'CheckboxTri [focused] [disabled]',
+  args: {
+    className: 'pseudo-focus-visible',
+    disabled: true,
+    defaultChecked: 'indeterminate',
+  },
+};
+
+const CheckboxTriControlled = (args: CheckboxTriArgs) => {
+  const shiftKeyRef = React.useRef(false);
+  const [checked, setChecked] = React.useState<CheckboxTriArgs['checked']>(args.defaultChecked ?? false);
+  
+  // XXX we could maybe turn this shift-click into a feature:
```

Comment:
XXX = Left over from some personal comment? Maybe just replace if for TODO

Reply:
"XXX" comments are a fairly common convention in code: https://stackoverflow.com/questions/1452934/what-is-the-meaning-of-xxx-in-code-comments

Could be a "TODO" also, but this isn't really something that needs to be done/fixed, more of a "hey here's a thought".